src/Aspect/ContractCheckerAspect.php

```diff
@@ -52,21 +49,7 @@ public function __construct(Reader $reader)
      */
     public function preConditionContract(MethodInvocation $invocation)
     {
-        $object = $invocation->getThis();
-        $args   = $this->getMethodArguments($invocation);
-        $scope  = $invocation->getMethod()->getDeclaringClass()->name;
-
-        foreach ($invocation->getMethod()->getAnnotations() as $annotation) {
-            if (!$annotation instanceof Contract\Verify) {
-                continue;
-            }
-
-            try {
-                $this->ensureContractSatisfied($object, $scope, $args, $annotation);
-            } catch (\Exception $e) {
-                throw new ContractViolation($invocation, $annotation->value, $e);
-            }
-        }
+        (new PreconditionContract($this->reader))->check($invocation);
```

Comment:
I don't like, that we create new instances of PreconditionContract on every method call. This is very expensive. You should avoid unnecessary object initialization and method calls for better performance.

One more suggest: if you want to separate this logic into smaller classes, then it will be better to define 3 separate aspects.